### ISSUE TITLE
dask.array.ravel_multi_index doesn't replicate numpy API

### ISSUE THREAD

**Issue by dougiesquire** [dask/dask#7580]
**What happened**:
`dask.array.ravel_multi_index` fails when passed a tuple of integer arrays (`AttributeError: 'tuple' object has no attribute 'map_blocks'`)

**What you expected to happen**:
The documentation (copied from the equivalent numpy function) indicates that the first input should be tuple of integer arrays, one array for each dimension.

**Minimal Complete Verifiable Example**:
```python
import numpy as np
import dask.array as dsa

arr = np.array([[3,6,6],[4,5,1]]) # Example from numpy docs
darr = dsa.from_array(arr)

np.ravel_multi_index((arr, arr), (7,7))    # This works
dsa.ravel_multi_index((darr, darr), (7,7)) # This doesn't
```

**Environment**:

- Dask version: 2021.04.0
- Python version: 3.8.8
- Operating System: Linux 4.12.14-95.57-default
- Install method (conda, pip, source): conda

**Comment by jrbourbeau** [comment]
Thanks for reporting @dougiesquire! Supporting a tuple of Dask arrays as input definitely seems in scope. cc'ing @Illviljan who added the `ravel_multi_index` implementation

**Comment by Illviljan** [comment]
I've got a potential fix over at #7584. Feel free to try it out.
